### DISCUSSION TITLE
Question - Answer,; Problem - Solution; likes of Theorem - Proof Environments

### DISCUSSION THREAD

**Discussion by aksigkvgithub** [quarto-dev/quarto-cli#9687] [Feature Requests]
### Description

In Quarto, there are few very good environments like Theorem - proof, Excercise - Solution, Example - Solution etc.

But, a teacher also requires such environments, as Question - Answer, and Problem - Solution, while preparing Assignment notes and Practical manuals. Out of these Solution environment is already there . However, Question, Answer, Problem environments are still missing. 

I tried many ways to do them by googling around but to no avail.

So, I request the developers to consider adding these environments in Quarto.

With regards,

Dr. A. K. Singh

**Comment by mcanouil** [comment]
The logic is the same as for https://github.com/quarto-ext/latex-environment and other extension.
Lua filters are still a good way to wrap some logics.

Anyhow, this is already tracked in the below issue, where it would be better to add related suggestions:

- https://github.com/quarto-dev/quarto-cli/issues/3979
